r-getting-started.Rmd

```diff
@@ -0,0 +1,329 @@
+easily share and run the code outside of the IDE.
+
+We have now told the computer what to do by using R code, and we have run that
+code in two different ways within our "workbench" - the RSudio IDE. Let's
+see what else we can use in this environment to help us.
+
+## Assigning and Recalling Objects {#r-getting-started-objects}
+
+Besides running functions that *do* something (as above), we'll also want
+to use R to keep track of information that we're using throughout
+our analysis. We save that information by creating an
+[object](glossary.html#object))
+using a name we select, special set of symbols known as the
+assignment operator (`<-`), and then the information to save.
+For example:
+
+```{r}
+message <- "It is a far, far better thing that I do, than I have ever done"
+name_length <- length("Sydney Carton")
+```
+
+On the top right pane, there's a
+tab that says "Environment". If you click on that, you'll see the
+objects you just created listed under "Values." Objects can represent data of
+various *types* - character (collection of letters and numbers encased in
+quotation marks), numeric (numbers, including decimals) and more.
+
+Referencing the name of an object by itself will print the data you assigned
+to the Console. You can also use objects as input to a function:
+
+```{r}
+message
+class(message)
+```
+
+These examples are objects representing small, fairly simple data.
+Objects can also be much more complex, representing large datasets.
+Hopefully you can imagine how assigning objects can allow you to
+work more easily with complex data, which you'll have a chance to do
+in the next section.
+
+## Viewing installed packages {#r-viewing-installed-packages}
+
+Functions in R are grouped together into collections of related code called
+packages. There is a tab in the lower right pane in RStudio called "Packages"
+that lists all of the packages installed in your version of R.
+Scrolling down the list, you'll note that some packages have the box on the
+left checked. This means the package has been loaded and is available for use.
+
+In addition to the built-in packages that come with every installation of R,
+we have the ability to use packages written by other programmers.
+One of the most common places to find such code is the
+[Comprehensive R Archive Network](https://cran.r-project.org), or CRAN.
+RStudio allows a straightforward way to download and install packages
+through CRAN. If you click on the "Install" button in the Packages tab,
+a window will appear allowing you to install from "Repository (CRAN)."
+In the space below "Packages," type "tidyverse." Making sure the box next to
+"Install dependencies" is checked, click "Install" to download and install
+this collection of code (which we'll be using in later sections).
+
+If you'd prefer to use code to install packages, rather than the RStudio pane,
+you could include the following in your script:
+
+```{r eval=FALSE}
+install.packages("tidyverse")
```

Comment:
This shouldn't be included in an R script. If they accidentally "source", then all of tidyverse will be installed again (or updated), which can be a pain in the ass.

Reply:
OK, I think I've clarified this regarding what to execute where.